core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -359,8 +359,17 @@ object UnitActions {
     }
 
     private fun takeOverTilesAround(unit: MapUnit) {
-        // one of the neighbour tile must belong to unit's civ, so nearestCity will be never `null`
-        val nearestCity = unit.currentTile.neighbors.first { it.getOwner() == unit.civInfo }.getCity()
+        // This method should only be called for a citadel - therefore one of the neighbour tile
+        // must belong to unit's civ, so minByOrNull will be never `null` (minBy is deprecated),
+        // and the getCity !! assertion is guaranteed by the filter selector.
+        // In the rare case more than one city owns tiles neighboring the citadel
+        // this will prioritize the nearest one not being razed
+        val nearestCity = unit.currentTile.neighbors
+            .filter { it.getOwner() == unit.civInfo }
+            .minByOrNull {
```

Comment:
I must say, this is highly unreadable. Maybe the minByOrNulll into a function, like 'fun getCityPriority(tileInfo)' or something. That wat you can also get rid of the if/else